packages/space-opera/src/components/mobile_view/mobile_view.ts

```diff
@@ -0,0 +1,216 @@
+
+  @internalProperty() gltfUrl: string|undefined;
+  @internalProperty() config: ModelViewerConfig = {};
+  @internalProperty() arConfig: ArConfigState = {};
+  @internalProperty() camera: Camera = INITIAL_CAMERA;
+  @internalProperty() hotspots: HotspotConfig[] = [];
+  @internalProperty() gltf?: GltfModel;
+
+  @internalProperty() pipeId = window.location.search.replace('?id=', '');
+  @internalProperty() base = 'https://ppng.io/modelviewereditor';
+  @internalProperty() snippetPipeUrl = `${this.base}-state-${this.pipeId}`;
+  @internalProperty() updatesPipeUrl = `${this.base}-updates-${this.pipeId}`;
+  @internalProperty() mobilePingUrl = `${this.base}-ping-${this.pipeId}`;
+
+  stateChanged(state: State) {
+    this.gltfUrl = getGltfUrl(state);
+    this.config = getConfig(state);
+    this.arConfig = getArConfig(state);
+    this.hotspots = getHotspots(state);
+    this.camera = getCamera(state);
+    this.gltf = getGltfModel(state);
+  }
+
+  getSrcPipeUrl(srcType: string): string {
+    return `https://ppng.io/modelviewereditor-srcs-${srcType}-${this.pipeId}`;
+  }
+
+  // TODO: https://javascript.info/fetch-progress
+  async waitForModel() {
+    await fetch(this.getSrcPipeUrl('gltf'))
+        .then(response => response.blob())
+        .then(blob => {
+          const modelUrl = URL.createObjectURL(blob);
+          reduxStore.dispatch(dispatchGltfUrl(modelUrl));
+          reduxStore.dispatch(dispatchSetHotspots([]));
+        })
+        .catch((error) => {
+          console.error('Error:', error);
+        });
+  }
+
+  async waitForState(envChanged: boolean) {
+    let partialState: any = {};
+    await fetch(this.snippetPipeUrl)
+        .then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error('Something went wrong');
+          }
+        })
+        .then((responseJson) => {
+          partialState = responseJson;
+        })
+        .catch((error) => {
+          console.log('error', error);
+        });
+
+    // These links would be corresponding to the original editor's link.
+    if (envChanged) {
+      partialState.config.environmentImage = undefined;
+    } else if (this.config.environmentImage) {
+      partialState.config.environmentImage = this.config.environmentImage;
+    }
+    partialState.config.src = this.gltfUrl;
+
+    reduxStore.dispatch(dispatchSetHotspots(partialState.hotspots));
+    reduxStore.dispatch(dispatchSetCamera(partialState.camera));
+    reduxStore.dispatch(dispatchSetConfig(partialState.config));
+    reduxStore.dispatch(dispatchArConfig(partialState.arConfig));
+  }
+
+  async waitForEnv(envIsHdr: boolean) {
+    await fetch(this.getSrcPipeUrl('env'))
+        .then(response => response.blob())
+        .then(blob => {
+          // simulating createBlobUrlFromEnvironmentImage
+          const addOn = envIsHdr ? '#.hdr' : '';
+          const envUrl = URL.createObjectURL(blob) + addOn;
```

Comment:
Looks like you're working around one of model-viewer's hacks here. We should really fix up how we recognize HDRs someday...